src/packagedcode/go_mod.py

```diff
@@ -31,10 +30,48 @@
 import re
 
 import attr
+from packageurl import PackageURL
+
 
+TRACE = False
 
+logger = logging.getLogger(__name__)
+
+if TRACE:
+    import sys
+    logging.basicConfig(stream=sys.stdout)
+    logger.setLevel(logging.DEBUG)
+
+
+@attr.s()
+class GoModules(object):
+    namespace = attr.ib(default=None)
+    name = attr.ib(default=None)
+    version = attr.ib(default=None)
+    module = attr.ib(default=None)
+
+    @property
+    def purl_gosum(self):
+        return PackageURL(
+                    type='golang',
+                    namespace=self.namespace,
+                    name=self.name,
+                    version=self.version
+                ).to_string()
+
+    @property
+    def purl_gomod(self):
+        return PackageURL(
+                    type='golang',
+                    namespace=self.namespace,
+                    name=self.name
```

Comment:
And here, gomod does not have version info?

Reply:
@majurg go.mod file does not contains the version of module. We need to find the another way to get the version info.

Reply:
No problem I just wanted to be sure.